### PR TITLE
PARQUET-1243: [C++] Throw more informative exception when reading a length-0 Parquet file

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

File: cpp/src/parquet/file_reader.cc [hunk 1]
```diff
@@ -161,8 +161,13 @@ class SerializedFile : public ParquetFileReader::Contents {
...
     int64_t file_size = source_->Size();
 
-    if (file_size < FOOTER_SIZE) {
-      throw ParquetException("Corrupted file, smaller than file footer");
+    if (file_size == 0) {
+      throw ParquetException("Invalid Parquet file size is 0 bytes");
+    } else if (file_size < FOOTER_SIZE) {
+      std::stringstream ss;
+      ss << "Invalid Parquet file size is " << file_size
+         << " bytes, smaller than standard file footer";
```
Comment (majetideepak) [inline cpp/src/parquet/file_reader.cc]: Should we print the standard file footer value as well?

Comment (wesm) [reply cpp/src/parquet/file_reader.cc]: Sure, will add

**Comment by pitrou** [comment]
This PR needs rebasing.

**Comment by wesm** [comment]
Rebased

**Comment by wesm** [comment]
rebased again to try to get a passing appveyor build

**Pull request closed without merging**